airflow/kubernetes/kubernetes_helper_functions.py

```diff
@@ -84,4 +84,8 @@ def annotations_to_key(annotations: Dict[str, str]) -> Optional[TaskInstanceKey]
             .scalar()
         )
 
+    if not run_id:
+        # TODO: Is this the right way of handling it?
+        raise ValueError("Run Id has to be defined either by `run_id` annotation "
+                         "or derived from `execution_Date")
```

Comment:
We can also just do

```python
(run_id,) = (
    session.query(TaskInstance.run_id)
    ...
    .one()
)
```

and let SQLAlchemy raise when there’s not a matching task instance found (it shouldn’t happen).